doc/developer/design/20210505_kafka_keys_in_sql.md

```diff
@@ -0,0 +1,165 @@
+# Kafka Keys as Values
+
+## Summary
+
+<!--
+// Brief, high-level overview. A few sentences long.
+// Be sure to capture the customer impact - framing this as a release note may be useful.
+-->
+
+Kafka is unique in our supported sources (although not all possible sources,
+e.g. [DynamoDB Streams][ddb]) in that it partitions messages into a Key part and
+a Value part. The Key is intended to be the equivalent of a primary key in a
```

Comment:
Nit: there are also timestamp and headers fields on each message.

Reply:
I intended those to come in as the more general "metadata" future work section. I'll rephrase this a bit.